pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -109,8 +109,13 @@ func (a *Assignment) ToAPI() []kueue.PodSetAssignment {
 func (a *Assignment) TotalRequestsFor(wl *workload.Info) resources.FlavorResourceQuantities {
 	usage := make(resources.FlavorResourceQuantities)
 	for i, ps := range wl.TotalRequests {
+		// in case of partial admission scale down the quantity
+		aps := a.PodSets[i]
+		if aps.Count != ps.Count {
+			ps = *ps.ScaledTo(aps.Count)
+		}
 		for res, q := range ps.Requests {
```

Comment:
```suggestion
		aps := a.PodSets[i]
		for res, q := range ps.ScaledTo(aps.Count).Requests {
```

nit: check whether there is a change of `pod count` in `ScaledTo`, and call ScaledTo in the `range` line

Reply:
I agree that moving the check into `ScaledTo` is preferred. It matches my expectations on how the function should behave (safe and performant).

Reply:
`ScaleTo` is now checking if the count is different but for it's own purpose (to skip the actual scaling), having it also skip the cloning it's odd for me.  As well as cloning when it's not needed.